helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -145,32 +145,28 @@ public int hashCode() {
 
   private class ZkPathStatRecord {
     private final String _path;
-    private Stat _stat = null;
-    private boolean _checked = false;
+    private AtomicBoolean _isChecked = new AtomicBoolean(false);
 
-    public ZkPathStatRecord(String path) {
+    ZkPathStatRecord(String path) {
       _path = path;
     }
 
-    public boolean pathExists() {
-      return _stat != null;
-    }
-
-    public boolean pathChecked() {
-      return _checked;
+    boolean isPathChecked() {
+      return _isChecked.get();
     }
 
     /*
      * Note this method is not thread safe.
      */
-    public void recordPathStat(Stat stat, OptionalLong notificationTime) {
-      _checked = true;
-      _stat = stat;
-
-      if (_monitor != null && stat != null && notificationTime.isPresent()) {
+    void recordPathStat(Stat stat, long notificationTime) {
+      if (isPathChecked()) {
+        return;
+      }
+      _isChecked.set(true);
+      if (_monitor != null && stat != null) {
         long updateTime = Math.max(stat.getCtime(), stat.getMtime());
-        if (notificationTime.getAsLong() > updateTime) {
-          _monitor.recordDataPropagationLatency(_path, notificationTime.getAsLong() - updateTime);
+        if (notificationTime > updateTime) {
+          _monitor.recordDataPropagationLatency(_path, notificationTime - updateTime);
```

Comment:
I did not see the value of changing this. If you feel this is necessary, please make it for separate PR as it does impact logic and purely code refactoring.

Reply:
The latest changes will not need to touch the existing codes too much, so it's no longer valid. Could you take another look, thanks?